notes/filters.py

```diff
@@ -0,0 +1,151 @@
+from taggit.models import Tag
+import django_filters
+from django import forms
+from .models import Note, Notebook
+
+class NoteFilter(django_filters.FilterSet):
+    notebook = django_filters.ModelChoiceFilter(
+        queryset=Notebook.objects.none(),
+        empty_label="All Notebooks"
+    )
+    tags = django_filters.CharFilter(method='filter_tags')
+    sort_by_created = django_filters.ChoiceFilter(
+        label='Sort by Created Time',
+        choices=[('created', 'Old to First'), ('-created', 'First to Old')],
+        method='filter_sort_by_created'
+    )
+    sort_by_modified = django_filters.ChoiceFilter(
+        label='Sort by Modified Time',
+        choices=[('modified', 'Old to First'), ('-modified', 'First to Old')],
+        method='filter_sort_by_modified'
+    )
+    sort_by_title = django_filters.ChoiceFilter(
+        label='Sort by Title',
+        choices=[('title', 'A to Z'), ('-title', 'Z to A')],
+        method='filter_sort_by_title'
+    )

```

Comment:
You can use OrderFilter to replace these 3 filters.
https://django-filter.readthedocs.io/en/latest/ref/filters.html#orderingfilter